### PR TITLE
Ecom API Support

### PR DESCRIPTION
**Pull request by bramski**
- Add ecom API to the available clients
- change package build to enable typescript
- refactor index.js to add named exports
- refactor and rename files to make sense with two API clients

### CODE REVIEW

File: src/utils/RetailApiCursor.js [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
-class ApiCursor extends Readable {
-  constructor(baseUrl, resource, instance, qs, opts) {
+class RetailApiCursor extends Readable {
+  constructor(baseUrl, resource, instance, queryString) {
     super();
     this._baseUrl = baseUrl;
     this._resource = resource;
     this._instance = instance;
-    this._qs = qs;
-    this.opts = opts;
```
Comment (bramski) [inline src/utils/RetailApiCursor.js:13]: Seems like opts was removed and `qs` was quite confusing so I gave the variable a better name.

File: src/utils/HttpClient.js [hunk 2]
```diff
@@ -1,75 +0,0 @@
-const https = require('https');
```
Comment (bramski) [inline src/utils/HttpClient.js:1]: This file was entirely unused so I removed it. Seems like you guys went with axios which is totally fine so you don't need this file.

**Comment by bramski** [comment]
@jcperez Any idea when you'll be able to review? I have a lot more changes waiting in the wings now but I'll issue another PR whence I'm done with those.

**Comment by bramski** [comment]
@andreadiazrepo Perhaps you can comment? I have a lot more to contribute. I have moved forward from this point significantly with enhancements to both the ecom and retail API for the project I'm working on.

**Comment by jcperez** [comment]
Hello @bramski - Sorry, we have been busy with some time-critical projects. I'm granting you write access to the project, so feel free to add more of your changes to this project. If you/your company uses Slack, and you have any questions, feel free to DM me  (juan at digitaport.com)

**Comment by bramski** [comment]
Thanks @jcperez! I've merged additional changes into the project. Feel free to have a look and release the library when you like. I've completed the project that needed this functionality so I suspect my continued contributions to this library will not continue.

**Pull request merged**